### PR TITLE
Add sc-component manager

### PR DESCRIPTION
**Pull request by MksmOrlov**
Add sc-component manager
use sc-component branch from https://github.com/ostis-ai/sc-component-manager/pull/67

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -73,6 +73,17 @@ To learn more about the platform, check out our [documentation](https://github.c
   ./scripts/install_platform.sh
   ```
 
+- Natively (using sc-component-manager)
+
+  ```sh
+  git clone https://github.com/ostis-ai/ostis-web-platform
+  cd ostis-web-platform/scripts
+  git checkout feature/component_manager
```
Comment (NikitaZotov) [inline README.md]:
```suggestion
```

File: README.md [hunk 2]
```diff
@@ -73,6 +73,17 @@ To learn more about the platform, check out our [documentation](https://github.c
...
+  ./install_platform_with_component_manager.sh
+  cd ../sc-machine/scripts
```
Comment (NikitaZotov) [inline README.md]:
```suggestion
```

File: README.md [hunk 3]
```diff
@@ -73,6 +73,17 @@ To learn more about the platform, check out our [documentation](https://github.c
...
+  ./run_sc_component_manager.sh -c ../../ostis-web-platform.ini
```
Comment (NikitaZotov) [inline README.md]:
```suggestion
  ./run_sc_component_manager.sh
```

File: scripts/install_platform_with_component_manager.sh [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+stage "Install ostis-web-platform"
+
+"${PLATFORM_PATH}/scripts/install_sc_machine_submodule.sh"
+"${PLATFORM_PATH}/scripts/install_sc_machine_dependencies.sh" --dev
+"${PLATFORM_PATH}/scripts/install_sc_component_manager_submodule.sh"
+"${PLATFORM_PATH}/scripts/build_sc_machine.sh"
+"${PLATFORM_PATH}/scripts/build_sc_component_manager.sh"
+"${PLATFORM_PATH}/scripts/build_kb.sh"
+
+stage "OSTIS-web-platform installed successfully"
```
Comment (NikitaZotov) [inline scripts/install_platform_with_component_manager.sh:21]: Please, add ci for check this script

File: scripts/set_vars.sh [hunk 5]
```diff
@@ -17,13 +17,23 @@ export SC_MACHINE_PATH="${SC_MACHINE_PATH:-${PLATFORM_PATH}/sc-machine}"
 export SC_WEB_REPO="${SC_WEB_REPO:-https://github.com/ostis-ai/sc-web.git}"
 export SC_WEB_BRANCH="${SC_WEB_BRANCH:-0.8.1-Unlock}"
 export SC_WEB_COMMIT="${SC_WEB_COMMIT:-""}"
-export SC_WEB_PATH="${SC_WEB_PATH:-${PLATFORM_PATH}/sc-web}"
+export SC_WEB_PATH="${SC_WEB_PATH:-${PLATFORM_PATH}/interface/sc-web}"
+
+export SC_COMPONENT_MANAGER_REPO="${SC_COMPONENT_MANAGER_REPO:-https://github.com/ostis-ai/sc-component-manager.git}"
```
Comment (NikitaZotov) [inline scripts/set_vars.sh:27]: Where ims.ostis.kb venvs were provided?

File: scripts/install_submodules.sh [hunk 6]
```diff
@@ -78,7 +43,9 @@ stage "Clone submodules"
 
 cd "${PLATFORM_PATH}" && git submodule update --init --recursive
 
-clone_project "${SC_MACHINE_REPO}" "${SC_MACHINE_PATH}" "${SC_MACHINE_BRANCH}" "${SC_MACHINE_COMMIT}"
-clone_project "${SC_WEB_REPO}" "${SC_WEB_PATH}" "${SC_WEB_BRANCH}" "${SC_WEB_COMMIT}"
+"${PLATFORM_PATH}/scripts/install_sc_machine_submodule.sh" "$@"
+"${PLATFORM_PATH}/scripts/install_sc_component_manager_submodule.sh" "$@"
+"${PLATFORM_PATH}/scripts/install_sc_web_submodule.sh" "$@"
+"${PLATFORM_PATH}/scripts/install_ims_kb_submodule.sh" "$@"
```
Comment (NikitaZotov) [inline scripts/install_submodules.sh:49]: Add, please, warning for installation of ims.ostis.kb

File: scripts/build_sc_component_manager.sh [hunk 7]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+set -eo pipefail
+
+if [ -z "${SC_COMPONENT_MANAGER_PATH}" ];
+then
+  source "$(cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd)"/../sc-component-manager/scripts/set_vars.sh
+fi
+
+export ROOT_CMAKE_PATH="${SC_COMPONENT_MANAGER_PATH}"
```
Comment (NikitaZotov) [inline scripts/build_sc_component_manager.sh:10]: Please, comment this setups to prevent removing by misunderstanding

**Review by NikitaZotov — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (MksmOrlov) [reply README.md]: why?

File: scripts/install_platform_with_component_manager.sh (hunk 4 above)
Comment (MksmOrlov) [reply scripts/install_platform_with_component_manager.sh:21]: ok, I will add it after merging https://github.com/ostis-ai/sc-component-manager/pull/67

File: scripts/install_submodules.sh (hunk 6 above)
Comment (MksmOrlov) [reply scripts/install_submodules.sh:49]: There is no alternative, that merged now

Comment (NikitaZotov) [reply scripts/install_submodules.sh:49]: Why? But you add sc-component-manager

File: README.md [hunk 8]
```diff
@@ -73,6 +73,15 @@ To learn more about the platform, check out our [documentation](https://github.c
   ./scripts/install_platform.sh
   ```
 
+- Natively (using sc-component-manager)
+
+  ```sh
+  git clone https://github.com/ostis-ai/ostis-web-platform
+  cd ostis-web-platform/scripts
+  ./install_platform_with_component_manager.sh
+  ./run_sc_component_manager.sh -c ../ostis-web-platform.ini
```
Comment (NikitaZotov) [inline README.md]: Why you need to pass this config? It passed in venvs

File: scripts/install_platform.sh [hunk 9]
```diff
@@ -14,6 +14,7 @@ stage "Install ostis-web-platform"
 "${PLATFORM_PATH}/scripts/install_submodules.sh"
 "${PLATFORM_PATH}/scripts/install_dependencies.sh" --dev
 "${PLATFORM_PATH}/scripts/build_sc_machine.sh"
+"${PLATFORM_PATH}/scripts/build_sc_component_manager.sh"
```
Comment (NikitaZotov) [inline scripts/install_platform.sh]: Is it not script to install platform without manager of components?

**Review by NikitaZotov — CHANGES_REQUESTED** [review]

File: scripts/install_platform_with_component_manager.sh (hunk 4 above)
Comment (NikitaZotov) [reply scripts/install_platform_with_component_manager.sh:21]: Let's merge this PR after marging PR https://github.com/ostis-ai/sc-component-manager/pull/67

File: scripts/install_submodules.sh (hunk 6 above)
Comment (MksmOrlov) [reply scripts/install_submodules.sh:49]: sc-component-manager is not deliting ims.ostis.kb. You can install it. The warning should be added after merging ostis-standard specification

File: scripts/install_platform.sh (hunk 9 above)
Comment (MksmOrlov) [reply scripts/install_platform.sh]: It is is not is it, yes

File: README.md (hunk 8 above)
Comment (MksmOrlov) [reply README.md]: image

Comment (NikitaZotov) [reply README.md]: By default, sc-server provide CONFIG_PATH from venvs. Can we make this for sc-component-manager?

**Review by NikitaZotov — APPROVED** [review]

**Review by MakarenkoAI — APPROVED** [review]

**Pull request merged**